pandas/core/missing.py

```diff
@@ -75,6 +75,80 @@ def mask_missing(arr, values_to_mask):
     return mask
 
 
+def update_array(this, that, overwrite=True, filter_func=None,
+                 errors='ignore'):
+    """
+    Update one array with non-NA values from another array.
+
+    Parameters
+    ----------
+    this : np.ndarray (one-dimensional)
+        The array being updated.
+    that : np.ndarray (one-dimensional)
+        The array being used to update.
+    overwrite : bool, default True
+        How to handle non-NA values for overlapping keys:
+
+        * True: overwrite original array's values with values from `that`.
+        * False: only update values that are NA in `this`.
+
+    filter_func : callable(1d-array) -> boolean 1d-array, optional
+        Can choose to replace values other than NA. Return True for values
+        that should be updated.
+    errors : {'raise', 'ignore'}, default 'ignore'
+        If 'raise', will raise a ValueError if `this` and `that` both contain
+        non-NA data in the same place.
+
+    Raises
+    ------
+    ValueError
+        When `errors='raise'` and there's overlapping non-NA data.
+
+    Returns
+    -------
+    updated : np.ndarray (one-dimensional)
+        The updated array.
+
+    See Also
+    --------
+    Series.update : Similar method for `Series`.
+    DataFrame.update : Similar method for `DataFrame`.
+    dict.update : Similar method for `dict`.
+    """
+    updated = _update_array(this, that, overwrite=overwrite,
+                           filter_func=filter_func, errors=errors)
+    return this if updated is None else updated
```

Comment:
@jreback 
You asked me to move the function here and make it public as `update_array`. This had the problem that it had the fallback of returning `None` in case there were no changes, which is necessary to avoid pointlessly updating the columns in `NDFrame.update`.

However, I was not comfortable with a public function having such unexpected behaviour, so I made the function I used originally private again (but still in `core.missing`) and just made a thin wrapper around it public, that only filters out the None-returning case.